etc/correctFieldMapJsonCase4.py

```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env python3
+'''
+correct field map json
+'''
+ 
+import os
+import sys
+import json
+import glob
+import collections
+
+def correctFieldMapJson(bids_dir,sub,ses=None):
+    
+    if ses: #ses not None
+        sub_prefix = '{}_{}'.format(sub,ses)
+        sub_path_prefix=os.path.join(sub,ses)
+        sub_root = '{}'.format(sub)
```

Comment:
I think this can just be `sub_root = sub` (or maybe `sub_root = str(sub)` if `sub` might not already be a string).

Actually, you may just want to get rid of this variable entirely because it seems to be the same as `sub` in all cases.